build/docker/Dockerfile

```diff
@@ -1,8 +1,9 @@
-FROM mcr.microsoft.com/dotnet/sdk:7.0.400-alpine3.17 AS build
-
+FROM mcr.microsoft.com/dotnet/sdk:7.0.401-cbl-mariner2.0-amd64 AS build
```

Comment:
Should we be pinning an architecture here vs using `mcr.microsoft.com/dotnet/sdk:7.0.401-cbl-mariner2.0`?

Reply:
Should instead of "sdk:7.0.401-cbl-mariner2.0-amd64" we use "sdk:7.0.401-alpine3.18"?

Reply:
Reference: https://hub.docker.com/_/microsoft-dotnet-nightly-sdk/

Reply:
@fhibf - we discussed mariner a couple weeks ago in standup. The idea is mariner should run better on Azure than alpine images since it's tuned. Mariner also has security tweaks which makes sense for healthcare industry.